src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -1263,29 +1301,55 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
 
-      WbNode *const fromNode = WbNode::findNode(fromNodeId);
       WbNode *const toNode = WbNode::findNode(toNodeId);
-
-      if (!dynamic_cast<WbTransform *const>(toNode)) {
+      if (!toNode) {
+        mRobot->warn(
+          tr("'wb_supervisor_node_%1_pose_tracking' called for an invalid node.").arg(enable ? "enable" : "disable"));
+        return;
+      }
+      WbTransform *const toTransformNode = dynamic_cast<WbTransform *>(toNode);
+      if (!toTransformNode) {
         mRobot->warn(tr("Node '%1' is not suitable for pose tracking, aborting request.").arg(toNode->modelName()));
         return;
       }
 
-      if (enable) {
-        WbTrackedPoseInfo trackedPose;
-        trackedPose.fromNode = dynamic_cast<WbTransform *>(fromNode);
-        trackedPose.toNode = dynamic_cast<WbTransform *>(toNode);
-        trackedPose.samplingPeriod = samplingPeriod;
-        trackedPose.lastUpdate = -INFINITY;
-        mTrackedPoses.append(trackedPose);
-      } else {
-        for (int i = 0; i < mTrackedPoses.size(); i++)
-          if (mTrackedPoses[i].fromNode == fromNode && mTrackedPoses[i].toNode == toNode) {
-            mTrackedPoses.removeAt(i);
-            break;
-          }
+      WbNode *const fromNode = WbNode::findNode(fromNodeId);
+      int index = -1;
+      for (int i = 0; i < mTrackedPoses.size(); i++) {
+        if (mTrackedPoses[i].fromNode == fromNode && mTrackedPoses[i].toNode == toTransformNode) {
+          index = i;
+          break;
+        }
       }
 
+      if (enable) {
+        WbTransform *const fromTransformNode = fromNode ? dynamic_cast<WbTransform *>(fromNode) : NULL;
+        if (fromNodeId && !fromTransformNode)
+          mRobot->warn(tr("Pose tracking can be exclusively used with Transform (or derived) 'from_node' argument, but '%1' is "
+                          "given. The absolute pose in global coordinates will be returned.")
+                         .arg(fromNode->modelName()));
+
+        if (index < 0) {
+          WbTrackedPoseInfo trackedPose;
+          trackedPose.fromNode = fromTransformNode;
+          trackedPose.toNode = toTransformNode;
+          trackedPose.samplingPeriod = samplingPeriod;
+          trackedPose.lastUpdate = -INFINITY;
+          mTrackedPoses.append(trackedPose);
+          if (fromTransformNode)
+            connect(fromTransformNode, &WbNode::destroyed, this, &WbSupervisorUtilities::removeTrackedPoseNode);
+          connect(toTransformNode, &WbNode::destroyed, this, &WbSupervisorUtilities::removeTrackedPoseNode);
+        } else {
+          mTrackedPoses[index].samplingPeriod = samplingPeriod;
+          mTrackedPoses[index].lastUpdate = -INFINITY;
+        }
+      } else if (index >= 0)
+        mTrackedPoses.removeAt(index);
+      else
+        mRobot->warn(tr("No active pose tracking could be found matching the node '%1' and 'from_node' argument.")
+                       .arg(toNode->modelName())
+                       .arg(fromNode->modelName()));
```

Comment:
If we use the `usefulName()` function we do not get the node type in case of a PROTO node.
So even if it will make it easier to identify which node is causing the issue, it won't give much information why the node is invalid.

By the way I used the same style as the other messages in the `WbSupervisorUtilities` class.
In all the other similar warning messages printed in the `WbSupervisorUtilities` class we always use `modelName()` and none of the messages is printing the `usefulName()`.